tests/test_asio.h

```diff
@@ -69,19 +72,27 @@ struct execution_context : asio::execution_context {
 
         template <typename Function>
         void dispatch(Function&& f, std::allocator<void>) const {
+            assert_has_impl();
             return impl_->dispatch(wrap_shared(std::forward<Function>(f)));
         }
 
         template <typename Function>
         void post(Function&& f, std::allocator<void>) const {
+            assert_has_impl();
             return impl_->post(wrap_shared(std::forward<Function>(f)));
         }
 
         template <typename Function>
         void defer(Function&& f, std::allocator<void>) const {
+            assert_has_impl();
             return impl_->defer(wrap_shared(std::forward<Function>(f)));
         }
 
+        void assert_has_impl() const {
+            if (!impl_)
+                throw std::logic_error("ozo::testing::execution_context::executor_type::assert_impl() no executor mock");
```

Comment:
Could you please place curly braces for if statement, like in project. The rationale for this are:
- changes inside `if ()` statement scope will not affect statement itself, so the history of the project will be more clear,
- sometimes weird  things like `if();` and similar are happened, so the braces help to avoid such thing.